### DISCUSSION TITLE
Query the search paths for `include` RC files

### DISCUSSION THREAD

**Discussion by bergercookie** [GothenburgBitFactory/taskwarrior#3258] [Q&A] [answered]
Hi there,

Currently in the task rc file users can include files, by name and taskwarrior will go look to include these files from pre-determined locations. I _think_ it's using the following code, in [Configuration::load](https://github.com/GothenburgBitFactory/libshared/blob/072e0e0cb91c1e05e02db59ed83924b43e992938/src/Configuration.cpp#L198-L252) to do that:

```
            // 2. Try path relative to the config file itself.
            if (!file_path.empty ())
            {
              std::string file_dir = Path(file_path).realpath();
              auto slash = file_dir.rfind('/');
              file_dir.resize (slash != std::string::npos ? slash + 1 : 0); // `/` is kept.
              Path file_relative (file_dir + included._data);
              if (file_relative.exists ())
              {
                included = file_relative;
                break;
              }
            }

            // 3. Try search paths.
            for (auto &search: search_paths)
            {
              Path concated (search + "/" + included._data);
              if (concated.exists ()) {
                included = concated;
                break;
              }
            }
            if (!included.exists ())
              throw format (
                  "Could not find file in CWD, directory of config file or search paths '{1}'.",
                  included._data);
          }
```

However, I'm having trouble figuring out how it knows what directories to look. 

So, my questions are:

* What's the contents of that `search_paths` variable ? At what part of the code is the search path being set?
* Is there a taskwarrior subcommand that I can query to figure out what's that search path so that I don't have to duplicate the search path logic in my application ?

For context I'm trying to solve the following issue https://github.com/ralphbean/taskw/pull/151 but as part of the newly created [bergercookie/taskw-ng](https://github.com/bergercookie/taskw-ng)

**Comment by bergercookie** [comment, marked as answer]
I think the only place the `::load` method is called is from Context.cpp

https://github.com/bergercookie/taskwarrior/blob/41608dc35fee0390d4679146034952fb12186053/src/Context.cpp#L470

```
  std::vector <std::string> searchPaths { TASK_RCDIR };
```

and TASK_RCDIR is set to `${CMAKE_INSTALL_PREFIX}/share/doc/task/rc` - let me know if this is not right.

Also, as far as I can tell, there's no way currently to query the TASK_RCDIR from a taskwarrior subcommand. Is there?

**Reply by djmitche** [reply]
I think the source code you've found is the definitive reference here.

Does that search path appear in `task diag`?

**Reply by bergercookie** [reply]
Nope, I don't see it. then I guess it's not suppoted atm. Thanks @djmitche
